nixos/modules/config/gtk/gtk-icon-cache.nix

```diff
@@ -0,0 +1,86 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+{
+  options = {
+    gtk.iconCache.enable = mkOption {
+      type = types.bool;
+      default = true;
```

Comment:
Okay, so this broke the installer tests and while it would be trivial to just add `gtk3` to the system closure, I think this should only default to `true` if the user really has a graphical setup, eg. not on a headless server system.

Reply:
We could set it to true in `xserver.nix`. I did something similar for the XDG modules, but I wasn't brave enough to make the default be false, because no doubt some people are running with weird systems but still want GTK icons.

Could we not just set it to false in the installer?

Reply:
Before this PR, in the `system-path` module, there was not an explicit dependency on `gtk3`, and the `hicolor` icon theme cache was being built with
```
          if [ -x $out/bin/gtk-update-icon-cache -a -f $out/share/icons/hicolor/index.theme ]; then
              $out/bin/gtk-update-icon-cache $out/share/icons/hicolor
          fi
```
That is, the cache was being built only if `gtk-update-icon-cache` was available on the system.

Would it be reasonable to keep this approach in the new `gtk` module? I am not sure.

Reply:
No, please don't go back to that, making the dependency explicit is the main reason I like this!

Reply:
Having this enabled by default also breaks NixOS on armv6l, because gtk eventually depends on valgrind, which doesn't build on armv6l.

Reply:
Probably the armv6 stuff is a separate issue which should be addressed in the appropriate place.